third_party/txt/src/txt/paragraph.cc

```diff
@@ -1135,34 +1182,104 @@ std::vector<Paragraph::TextBox> Paragraph::GetRectsForRange(
       // Ignore line max height and width and generate tight bounds.
-      boxes.insert(boxes.end(), kv.second.begin(), kv.second.end());
-    } else {
-      // Set each box to the max height of each line to ensure continuity.
-      float min_top = DBL_MAX;
-      float max_bottom = 0;
-      for (const Paragraph::TextBox& box : kv.second) {
-        min_top = std::min(box.rect.fTop, min_top);
-        max_bottom = std::max(box.rect.fBottom, max_bottom);
+      boxes.insert(boxes.end(), kv.second.boxes.begin(), kv.second.boxes.end());
+    } else if (rect_height_style == RectHeightStyle::kMax) {
+      for (const Paragraph::TextBox& box : kv.second.boxes) {
+        boxes.emplace_back(
+            SkRect::MakeLTRB(
+                box.rect.fLeft,
+                line_baselines_[kv.first] - line_max_ascent_[kv.first],
+                box.rect.fRight,
+                line_baselines_[kv.first] + line_max_descent_[kv.first]),
+            box.direction);
+      }
+    } else if (rect_height_style ==
+               RectHeightStyle::kIncludeLineSpacingMiddle) {
+      SkScalar adjusted_bottom =
+          line_baselines_[kv.first] + line_max_descent_[kv.first];
+      if (kv.first < line_ranges_.size() - 1) {
+        adjusted_bottom += (line_max_spacings_[kv.first + 1] -
+                            line_max_ascent_[kv.first + 1]) /
+                           2;
+      }
+      SkScalar adjusted_top =
+          line_baselines_[kv.first] - line_max_ascent_[kv.first];
+      if (kv.first != 0) {
+        adjusted_top -=
+            (line_max_spacings_[kv.first] - line_max_ascent_[kv.first]) / 2;
+      }
+      for (const Paragraph::TextBox& box : kv.second.boxes) {
+        boxes.emplace_back(SkRect::MakeLTRB(box.rect.fLeft, adjusted_top,
+                                            box.rect.fRight, adjusted_bottom),
+                           box.direction);
+      }
+    } else if (rect_height_style == RectHeightStyle::kIncludeLineSpacingTop) {
+      for (const Paragraph::TextBox& box : kv.second.boxes) {
+        SkScalar adjusted_top =
+            kv.first == 0
+                ? line_baselines_[kv.first] - line_max_ascent_[kv.first]
+                : line_baselines_[kv.first] - line_max_spacings_[kv.first];
+        boxes.emplace_back(
+            SkRect::MakeLTRB(
+                box.rect.fLeft, adjusted_top, box.rect.fRight,
+                line_baselines_[kv.first] + line_max_descent_[kv.first]),
+            box.direction);
       }
-      for (const Paragraph::TextBox& box : kv.second) {
-        boxes.emplace_back(SkRect::MakeLTRB(box.rect.fLeft, min_top,
-                                            box.rect.fRight, max_bottom),
+    } else {  // kIncludeLineSpacingBottom
+      for (const Paragraph::TextBox& box : kv.second.boxes) {
+        SkScalar adjusted_bottom =
+            kv.first >= line_ranges_.size() - 1
+                ? line_baselines_[kv.first] + line_max_descent_[kv.first]
```

Comment:
Remove the ternary operator here (similar to the change made to `adjusted_bottom` above)